assets/js/log-viewer/store.ts

```diff
@@ -17,16 +17,37 @@ interface LogStore {
   formattedLogLines: string;
   addLogLines: (newLogs: LogLine[]) => void;
   highlightedRanges: { start: number; end: number }[];
+  desiredLogLevels: string[] | undefined;
+  setDesiredLogLevels: (desiredLogLevel: string[] | undefined) => void;
 }
 
-function findSelectedRanges(logs: LogLine[], stepId: string | undefined) {
+// check if a log matches the desired log level
+function matchesLogFilter(
+  log: LogLine,
+  desiredLogLevels: string[] | undefined
+): boolean {
+  if (!desiredLogLevels) return true;
```

Comment:
For the sake of readability I would validate undefined explicitly because of the way this if sounds (not desired => true).